src/main/kotlin/org/rust/cargo/runconfig/RustBacktraceFilter.kt

```diff
@@ -1,17 +1,137 @@
+ * - Turn source code links into hyperlinks.
+ * - Dims funcion hashcodes to remove noise.
  */
 class RustBacktraceFilter(
     project: Project,
     cargoProjectDir: VirtualFile
-) : RegexpFileLinkFilter(
-    project,
-    cargoProjectDir,
-    "^\\s+at ${RegexpFilter.FILE_PATH_MACROS}:${RegexpFilter.LINE_MACROS}$") {
+) : Filter {
+    private val sourceLinkFilter = RegexpFileLinkFilter(project, cargoProjectDir, "^\\s+at ${RegexpFilter.FILE_PATH_MACROS}:${RegexpFilter.LINE_MACROS}$")
+    private val backtraceItemFilter = RustBacktraceItemFilter(project)
+
+    override fun applyFilter(line: String, entireLength: Int): Filter.Result? {
+        backtraceItemFilter.applyFilter(line, entireLength)?.let { return it }
+        sourceLinkFilter.applyFilter(line, entireLength)?.let { return it }
+        return null
+    }
+}
+
+/**
+ * Adds hyperlinks to function names in backtraces
+ */
+private class RustBacktraceItemFilter(
+    val project: Project
+) : Filter {
+    private val pattern = Pattern.compile("^(\\s*\\d+:\\s+0x[a-f0-9]+ - )(.+)(::h[0-9a-f]+)$")!!
+    private val docManager = PsiDocumentManager.getInstance(project)
+
+    override fun applyFilter(line: String, entireLength: Int): Filter.Result? {
+        val matcher = pattern.matcher(line)
+        if (!matcher.find()) return null
+        val funcName = matcher.group(2)
+        val resultItems = ArrayList<Filter.ResultItem>(2)
+
+        // Add hyperlink to the function name
+        val funcStart = entireLength - line.length + matcher.group(1).length
+        if (SKIP_PREFIXES.none { funcName.startsWith(it) }) {
+            extractFnHyperlink(funcName, funcStart)?.let { resultItems.add(it) }
+        }
+
+        // Dim the hashcode
+        val funcEnd = funcStart + funcName.length
+        resultItems.add(Filter.ResultItem(funcEnd, funcEnd + matcher.group(3).length, null, DIMMED_TEXT))
+
+        return Filter.Result(resultItems)
+    }
+
+    private fun extractFnHyperlink(funcName: String, start: Int): Filter.ResultItem? {
+        val func = ElementResolver.resolve(funcName, project) ?: return null
+        val funcFile = func.element.containingFile
+        val doc = docManager.getDocument(funcFile) ?: return null
+        val link = OpenFileHyperlinkInfo(project, funcFile.virtualFile, doc.getLineNumber(func.element.textOffset))
+        return Filter.ResultItem(start, start + funcName.length, link, func.pkg.isExternal)
+    }
+
+    private companion object {
+        val DIMMED_TEXT = EditorColorsManager.getInstance().globalScheme
+            .getAttributes(TextAttributesKey.createTextAttributesKey("org.rust.DIMMED_TEXT"))!!
+        val SKIP_PREFIXES = arrayOf(
+            // TODO: Put the whole std:: and core:: here?
+//            "std::sys::backtrace::",
+//            "std::panicking::",
+            "std::rt::lang_start"
+        )
+    }
+}
+
+
+private object ElementResolver {
+    private val vfm = VirtualFileManager.getInstance()
```

Comment:
This filed of an object (~ static field) seems a bit dangerous. It is lazily evaluated the first time this object is accessed, while `VirtualFileManager` is available only after the application has been initialized. Let's just query it inside the method.